File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        int n=intervals.size();

        vector<vector<int>>ans;

        for(int i=0;i<n;i++)
        {
            
            vector<int>a;

            int startInt=intervals[i][0];
            int endInt=intervals[i][1];

            for(int j=i+1;j<n;j++)
            {
                if(endInt>=intervals[j][0])
                {
                    endInt=max(endInt,intervals[j][1]);
                    i++;

                }
                else {
                    break;
                }
            }
            a.push_back(startInt);
            a.push_back(endInt);

            ans.push_back(a);
        }

        return ans;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """

        intervals.sort()

        st=intervals[0][0]
        end=intervals[0][1]

        ans=[]

        n=len(intervals)
        for i in range(1,n):
            el=intervals[i]
            if(el[0]>end):
                #it will add in it
                lis=[]
                lis.append(st)
                lis.append(end)
                ans.append(lis)
                st=el[0]
                end=max(end,el[1])
            else:
                end=max(end,el[1])
        
        lis=[]
        lis.append(st)
        lis.append(end)
        ans.append(lis)

        return ans


                


            
        